abi-bindings/go/packer/packer_test.go

```diff
@@ -0,0 +1,204 @@
+		for _, obj := range pkg.Types.Scope().Names() {
+			if obj == interfaceName {
+				ifaceObj := pkg.Types.Scope().Lookup(obj)
+				if ifaceObj == nil {
+					continue
+				}
+				if typ, ok := ifaceObj.Type().Underlying().(*types.Interface); ok {
+					iface = typ
+					break
+				}
+			}
+		}
+	}
+	require.NotNilf(t, iface, "Interface %s not found in package %s", interfaceName, currentPackagePath)
+
+	pkgs, err := packages.Load(cfg, packagePath)
+	require.NoError(t, err)
+
+	allImplementers := []string{}
+
+	for _, pkg := range pkgs {
+		for _, file := range pkg.Syntax {
+			// Inspect types in the package
+			for _, decl := range file.Decls {
+				genDecl, ok := decl.(*ast.GenDecl)
+				if !ok {
+					continue
+				}
+				for _, spec := range genDecl.Specs {
+					// We have a Type spec
+					typeSpec, ok := spec.(*ast.TypeSpec)
+					if !ok {
+						continue
+					}
+					// Find object matching the TypeSpec name in the package scope
+					obj := pkg.Types.Scope().Lookup(typeSpec.Name.Name)
+					if obj == nil {
+						continue
+					}
+					// Check if the typed object we have is a struct
+					_, ok = obj.Type().Underlying().(*types.Struct)
+					if !ok {
+						continue
+					}
+					// Lookup the object by name and cast it to types.Named
+					typeName := typeSpec.Name.Name
+					typeObj := pkg.Types.Scope().Lookup(typeName)
+					if typeObj == nil {
+						continue
+					}
+					named, ok := typeObj.Type().(*types.Named)
+					if !ok {
+						continue
+					}
+					// Finally check if the named struct we have implements the interface
+					if types.Implements(types.NewPointer(named), iface) {
+						allImplementers = append(allImplementers, named.Obj().Name())
+					}
+				}
+			}
+		}
+	}
+	return allImplementers
+}
+
+func TestExhaustivePacking(t *testing.T) {
+	implementers := findAllImplementers(t)
+	for _, structName := range implementers {
+		packerType, ok := packerTypes[structName]
+		require.True(
+			t,
+			ok,
+			fmt.Sprintf("Struct %s not found in packer_test.go packerTypes map. Add it to check for exhaustiveness", structName),
+		)
+
+		randomizedStruct, err := randomizeStruct(packerType)
+		require.NoError(t, err)
+
+		v1 := randomizedStruct.(ABIPacker)
+		encoded, err := v1.Pack()
+		require.NoError(t, err)
+		v2 := reflect.New(reflect.TypeOf(v1).Elem()).Interface().(ABIPacker)
+		err = v2.Unpack(encoded)
+		require.NoError(t, err)
+		require.Equal(t, v1, v2)
+	}
+}
+
+func randomizeStruct(packerStruct interface{}) (interface{}, error) {
+	v := reflect.ValueOf(packerStruct).Elem()
+
+	for i := 0; i < v.NumField(); i++ {
+		field := v.Field(i)
+		err := randomizeField(field)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return packerStruct, nil
+}
+
+func randomizeField(field reflect.Value) error {
+	fieldType := field.Type()
+	switch fieldType.Kind() {
+	// The list only includes types supported by abigen
+	case reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64:
+		field.SetInt(rand.Int63())
```

Comment:
I'm not sure how ints are capped, but it would be clearer to read if we instead bounded the random value to be within the smallest integer's range, or else hardcoded to a static value also within that range.

Reply:
> None of this matters since we don't rely on randomness for the validity of the test

Good enough for me!